File: tkpdtransaction/src/main/java/com/tokopedia/transaction/checkout/data/repository/MultipleAddressRepository.java
package com.tokopedia.transaction.checkout.data.repository;

import com.google.gson.JsonObject;

import rx.Observable;

/**
 * Created by kris on 2/5/18. Tokopedia
 */

public class MultipleAddressRepository implements IMultipleAddressRepository{

    public MultipleAddressRepository() {
    }

    @Override
    public Observable<String> sendMultipleAddressData(JsonObject dataToSend) {
        return null;
    }
}

File: features/transaction/purchase_platform/src/main/java/com/tokopedia/purchase_platform/features/checkout/subfeature/multiple_address/data/repository/MultipleAddressRepository.kt
package com.tokopedia.purchase_platform.features.checkout.subfeature.multiple_address.data.repository

import com.tokopedia.purchase_platform.features.checkout.subfeature.multiple_address.data.api.MultipleAddressApi
import com.tokopedia.purchase_platform.features.checkout.subfeature.multiple_address.data.model.response.CartMultipleAddressDataListResponse
import rx.Observable
import javax.inject.Inject

/**
 * Created by Irfan Khoirul on 2019-08-30.
 */

class MultipleAddressRepository @Inject constructor(val multipleAddressApi: MultipleAddressApi): IMultipleAddressRepository {

    override fun getCartMultipleAddressList(param: Map<String, String>): Observable<CartMultipleAddressDataListResponse> {
        return multipleAddressApi.getMultipleAddressCartList(param).map { cartResponseResponse ->
            cartResponseResponse.body()?.convertDataObj(CartMultipleAddressDataListResponse::class.java)
        }
    }

}